packages/gatsby/src/utils/adapter/get-route-path.ts

```diff
@@ -11,7 +11,9 @@ function maybeDropNamedPartOfWildcard(
   return path.replace(/\*.+$/, `*`)
 }
 
-export function getRoutePathFromPage(page: IGatsbyPage): string {
+export function getRoutePathFromPage(
+  page: Pick<IGatsbyPage, "path" | "matchPath">
```

Comment:
this (and other places I applied `Pick<>` to `IGatsbyPage`) was mainly to figure out minimal subset of `IGatsbyPage` fields that is actually needed for engines to work.

After everything was narrowed down this allowed me to produce final subset of fields to be inlined in engine ( https://github.com/gatsbyjs/gatsby/pull/38974/files#diff-6e15aa2b28579972ae31b662310cd05809325fe178e51e48e4a89bbbc3b34134R45-R54 )